### ISSUE TITLE
Handling multiple OutputSerializers

### ISSUE THREAD

**Issue by jsandovalc** [HackSoftware/Django-Styleguide#67]
Hi,

A general use case for the product I'm building is returning different information for an endpoint depending on the type of user querying it. Before adopting this styleguide we handled this by creating different serializers and using them according to the user making the request.

How do you handle this use case?. Creating multiple OutputSerializers inside the view kind of pollutes it. Or maybe some kind of dynamic serializer is recommended?.

Thanks for the great work.

**Comment by RadoRado** [comment]
@jsandovalc Hello & thanks for the question.

The way you've described it sounds like you can approach things as follows:

```
class SomeApi(APIView):
    def get(self, request):
        user = request.user
        data = some_user_related_selector(user)
        return Response(data)
```

Now, how you handle things in `some_user_related_selector` is up to you. If you are returning 4-5 fields for the user, I'd even go as far as just returning a plain dict & skipping serializers altogether.

If things are more complex, you can easily define different serializers to use for the different user types & use them in the selector. 

If the case is simple enough (2 types of users), you can define those in the API & just use a simple if.

tl;dr - figure out what's best for your case & level of complexity & go with it.

Cheers :v:

**Comment by jsandovalc** [comment]
Thanks for your answer!!
